generator/src/main/java/com/scottlogic/deg/generator/constraints/StringConstraintsCollection.java

```diff
@@ -0,0 +1,180 @@
+            if (longerThan >= shorterThan - 1){
+                return Optional.of(true); //field1 < 10 & field1 > 20 | field1 < 10 & field1 > 10
+            }
+
+            return Optional.of(false);
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<Boolean> isContradictoryToMinLength(int longerThan, AtomicConstraint otherConstraint) {
+        if (otherConstraint instanceof IsStringShorterThanConstraint){
+            int shorterThan = ((IsStringShorterThanConstraint) otherConstraint).referenceValue;
+            if (shorterThan <= longerThan){
+                return Optional.of(true); //field1 > 20 & field1 < 20 | field1 > 10 & field1 < 10
+            }
+
+            return Optional.of(false);
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<Boolean> isContradictoryToOfLengthConstraint(int requiredLength, AtomicConstraint otherConstraint) {
+        if (otherConstraint instanceof IsStringLongerThanConstraint){
+            int longerThan = ((IsStringLongerThanConstraint) otherConstraint).referenceValue;
+            if (requiredLength <= longerThan){
+                return Optional.of(true); // field1 ofLength 10 & field1 > 10 (i.e field1 >= 11)
+            }
+
+            return Optional.of(false); //technically non-contradictory
+        }
+
+        if (otherConstraint instanceof IsStringShorterThanConstraint){
+            int shorterThan = ((IsStringShorterThanConstraint) otherConstraint).referenceValue;
+            if (requiredLength >= shorterThan){
+                return Optional.of(true); // field1 ofLength 10 & field1 < 10 (i.e field1 <= 9)
+            }
+
+            return Optional.of(false); //technically non-contradictory
+        }
+
+        return Optional.empty();
+    }
+
+    private AtomicConstraint complementOfConstraint(AtomicConstraint negatedConstraint) {
+        if (negatedConstraint instanceof IsStringShorterThanConstraint){
+            IsStringShorterThanConstraint shorterThan = (IsStringShorterThanConstraint) negatedConstraint;
+            return new IsStringLongerThanConstraint(
+                shorterThan.field,
+                shorterThan.referenceValue - 1, //field1 < 10 --> field1 > 9
+                shorterThan.getRules());
+        }
+
+        if (negatedConstraint instanceof IsStringLongerThanConstraint){
+            IsStringLongerThanConstraint longerThan = (IsStringLongerThanConstraint) negatedConstraint;
+            return new IsStringShorterThanConstraint(
+                longerThan.field,
+                longerThan.referenceValue + 1, //field1 > 9 --> field1 < 10
+                longerThan.getRules());
+        }
+
+        if (negatedConstraint instanceof StringHasLengthConstraint){
+            StringHasLengthConstraint ofLength = (StringHasLengthConstraint) negatedConstraint;
+
+            //can use either longerThan <length> or shorterThan <length>
```

Comment:
**needs** to test both. as this will claim that 
"not length 10"
is contradictory to 
"length 11"

Reply:
Good spot, additional tests created